File: Learning/sort/OptimizedBubblesort.java
public class OptimizedBubblesort {
    public static void main(String args[]) {
        System.out.println("Hello World!");
        int arr[] ={12,14,3,5,7,16};
        bublesort(arr);
        printArray(arr);
        System.out.print("Thanks");
    }
    public static void bublesort(int[] arr) {
        int i,j;
         boolean swapped;
        for( i=0;i<arr.length;i++){
            swapped=false;
             System.out.print(":Inside bubble sort");
            for( j=0;j<arr.length-i-1;j++){     
                if(arr[j]>arr[j+1]){                    
                    int temp = arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;     
                    swapped = true;  
              }
            }
            if(swapped = false){
                break;
            }

        }
    }
    
  public static void printArray(int arr[]) 
    { 
        int n = arr.length; 
        for (int i=0; i<n; ++i) 
            System.out.print(arr[i] + " "); 
        System.out.println(); 
    } 

}

File: Learning/sort/optimizedBubbleSort.js
const numbers = [99,2,3,1,234,11,33,55,23,232];

function bubbleSort(array) {
    const length = array.length;
    let swapped;
    for(let i=0; i<length-1; i++) {
        swapped = false;
        for(let j=0;j<length-i-1;j++) {
            if(array[j] > array[j+1]) {
                //swap
                let temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;
                swapped = true;
            }
        }
        if(swapped = false) {
            break;
        }
    }
}

bubbleSort(numbers);
console.log(numbers);